/*
	CF 通用对接协议
	RPC名称:cf.rpc
	Exchange:cf.notify
*/
package cf;

message Header{
	required string sender=1;
	required string sender_type=2;
	optional uint32 invoke_id=3;
}
//-------------------------全部消息
message Message{
	// 头部
	required Header header=1;
	// 主体
	oneof content{
		// 请求:发送文本消息
		SendTextRequest req_send_text=10;
		// 回应:发送文本消息
		SendTextResponse res_send_text=11;
		// 通知:收到文本消息
		RecvTextNotify recv_text_notify=12;
  }
}
//-------------------------枚举值
// 数据源类型
enum ESourceType
{
	SMS=1;
	WECHAT=2;
	ALI=3;
	EMAIL=4;
	OTHER=10;
}
// 内容类型
enum EContentFormat
{
	TXT=1;
	HTML=2;
	JSON=3;
	XML=4;
	BASE64=5;
}

// 客户检索条件
message CustomerSearchCondition
{
	// 客户ID
	optional string id=10;
	// 微信相关
	optional string cf_account_id=11;
	optional string wechat_open_id=12;
	optional string wechat_union_id=13;
	optional string code=18;
	// Ali账号
	optional string ali_id=14;
	// 手机号码
	optional string mobile=15;
	// 设备MAC地址
	optional string device_mac=16;
	// Email地址
	optional string email=17;
}
//-------------------------内容定义

//-------------------------消息定义
// 请求:发送文本消息
message SendTextRequest
{
	// 发送账户ID
	required string account_id=10;
	// 发送目标
	repeated CustomerSearchCondition customer_search_condition=11;
	// 发送内容
	required string content=12;
	// 发送时段
	//optional TimeHorizon time_horizon=13;
}
// 回应:发送文本消息
message SendTextResponse
{
	// 错误码
	required int32 errcode=1;
	// 错误消息
	optional string errmsg=2;
	// 发送结果
	repeated int32 result=10;
}
// 通知:收到文本消息
message RecvTextNotify
{
	// 数据来源类型
	required ESourceType source_type=10;
	// 接收账户
    required string account_id=11;
	// 发送者
	required CustomerSearchCondition sender=12;
	// 文本内容
	required string content=13; 
}